cmd/ipfs/main.go

```diff
@@ -473,6 +481,9 @@ func (i *cmdInvocation) setupInterruptHandler() {
 	sig := allInterruptSignals()
 
 	go func() {
+		// wait till the context is ready to be closed
+		<-ctx.ContextIsReadyToBeClosed
```

Comment:
because we send to `ctx.ContextIsReadyToBeClosed` after bootstrapping (which can take seconds) we cannot kill the daemon during bootstrapping.

Perhaps we can instead use it like a variable, something like: 

```go
close(ctx.initDone)
```

and below, we can do:

```go
<-sig

// if we're still initializing, cannot use `ctx.GetNode()`
select {
default: // initialization not done
  fmt.Println("Received interrupt signal, shutting down...")
  os.Exit(-1)  // or maybe we want n.Close(). not sure.
  return
case <-initDone:
}

n, err := ctx.GetNode()
...
```